### ISSUE TITLE
Content applicability extent CRS

### ISSUE THREAD

**Issue by ccrook** [opengeospatial/CRS-Gridded-Geodetic-data-eXchange-Format#61]
Currently the CRS of the content applicability extent is not defined in GGXF.  There may have been a suggestion that it is always EPSG:4326 at one time, but I am not sure of that. 

The content applicability extent is represented in the NetCDF encoding by ACDD conventional attributes geospatial_lon_min, geospatial_lon_max, geospatial_lat_min, geospatial_lat_max.

The ACDD conventions specify EPSG:4326 (WGS84 lat/lon 2d geographic CRS) as a default but permit over-riding with an alternative CRS defined as by a geospatial_bounds_crs attribute (which is recommended to be an EPSG code).  

While I can't immediately see any explicit constraint in ACDD on the CRS being a 2d geographic CRS this is implied by the name of the attributes geospatial_lat_min/max and geospatial_lon_min/max.  The GGXF attributes are also named as latitudes and longitudes, so this again suggests that the CRS is constrained to be a 2d geographic CRS.

The GGXF specification should at least clarify the CRS of the content applicability extents in order for them to be usable at all.  I think it makes sense to follow ACDD and permit specification.  ACDD is ambiguous about the content of the geospatial_bounds_crs attribute - it recommends providing an EPSG code, but doesn't preclude a full WKT definition as the other CRSs in GGXF.  

For consistency with the rest of GGXF I believe GGXF should specify a crsWkt attribute of the contentApplicabilityExtent which is copied to an ACDD geospatial_bounds_crs attribute in the NetCDF encoding.  This could be defined as an optional attribute, but if so I think GGXF should specify the default value.  The GGXF specification should also include the constraint that the content applicability extent CRS is a 2d geographic CRS (unless we think otherwise).

**Comment by RogerLott** [comment]
GGXF document section 5.9.1 says "The description uses the Extent provisions from ISO 19115-1". In ISO 19115 (Geographic Information - Metadata) the definition of the geographic bounding box element of Extent includes a note "NOTE This is only an approximate reference so specifying the coordinate reference system is unnecessary and need only be provided with a precision of up to two decimal places". In ISO 19115 Extent is discovery metadata. By stating that GGXF uses the provisions of ISO 19115-1 arguably there is an implication that Extent in GGXF is also considered discovery metadata.

ACDD is more specific in defaulting explicitly to WGS 84. It may have taken this from the policy adopted for the EPSG Dataset. EPSG does not permit an alternative to WGS 84 to be declared. When the expected resolution of bounding box coordinates is 0.01 degree (~ 1 km) any geodetic [geographic] CRS will do.

Although nothing to do with extent, the OGC web API web specification for features also defaults to WGS 84 and, like ACDD, optionally permits alternatives to be explicitly stated.  Alternatives have to be identified through an OGC URI string which includes an authority and that authority's identifier. The OGC API features use case is to permit the exchange of data in CRSs other than WGS 84 (and these other CRSs do not have to be geographic).

@ccrook said "Note also that the current specification is inconsistent in that the extentVertical includes an optional extentVerticalCrsWkt attribute". This comes directly from ISO 19115-1 where it is needed to allow specification of vertical max/min limits either through heights or through depths. Currently GGXF is consistent with ISO 19115-1. 

extentVerticalCrsWkt is not optional. extentVertical is an optional element of Extent. But when extentVerticalMinimum and extentVerticalMaximum attributes are present, the inclusion of extentVerticalCrsWkt becomes mandatory.

The bounding box attributes westBoundLongitude, eastBoundLongitude, southBoundLatitude and northtBoundLatitude also come directly from ISO 19115-1. These are the only attributes of a geographic bounding box so it can only be 2D. However in 19115-1 a boundingPolygon can be of any dimension so it should not be assumed that a single alternative identified CRS could be used for both geographicBoundingBox and boundingPolygon - GGXF would need to either require both geographicBoundingBox and boundingPolygon to be in the same CRS (i.e. both be geographic 2D) or [more likely] to make provision for boundingBoxCrs and boundingPolygonCrs.

@ccrook also said "The GGXF specification should at least clarify the CRS of the content applicability extents in order for them to be usable at all". Without any clarification the Extent information currently is useable - for discovery metadata purposes. What other use cases need to be supported?

**Comment by RogerLott** [comment]
ACDD geospatial_bounds maps to GGXF boundingPolygon.  ACDD geospatial_bounds_CRS applies to geospatial_bounds, but not to bounding box.

**Comment by ccrook** [comment]
@RogerLott . Thanks, this is good clarification.  Following from this I wonder if it would be useful to add a clarification in the specification about the usage if contentApplicabilityExtent.  i.e. it serves as discovery metadata but not for testing whether the GGXF is applicable at a specific interpolation CRS coordinate.  

Following from that, do we need an option for defining at a file or group level what the actual valid extent of the file or group is, given that the accuracy and CRS of the contentApplicabilityExtent is undefined and may not be easily converted to interpolation CRS coordinates?  This also links to issue #58

**Comment by RogerLott** [comment]
Already in there. First para of 5.9.1 says "The spatial and, if relevant, temporal extent of all data in a GGXF file is given in the file header as discovery metadata".

**Comment by ccrook** [comment]
Yes, it is defined as discovery metadata.  I was thinking more of the second part of the usage statement above, which is what tripped up the ESRI developers.  Effectively it is a disclaimer, ie that the contentApplicabilityExtent is approximate -  it may not match the extents of the gridded data, and cannot be used to reliably determine if the content is applicable at a specific interpolation CRS coordinate.
